src/crab/type_domain.cpp

```
@@ -270,4 +256,782 @@ std::ostream& operator<<(std::ostream& os, const TypeGroup ts) {
+    return get_typeset(reg_type(r)).is_subset_of(to_typeset(TypeGroup::pointer));
+}
+
+bool TypeDomain::type_is_number(const Reg& r) const {
+    return get_typeset(reg_type(r)) == TypeSet::singleton(T_NUM);
+}
+
+bool TypeDomain::type_is_not_stack(const Reg& r) const { return !get_typeset(reg_type(r)).contains(T_STACK); }
+
+bool TypeDomain::type_is_not_number(const Reg& r) const { return !get_typeset(reg_type(r)).contains(T_NUM); }
+
+std::vector<TypeEncoding> TypeDomain::iterate_types(const Reg& reg) const {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)